### PR TITLE
Kds 1757 migrate contextmodal

### PR DESCRIPTION
**Pull request by gyfchong**
ContextModal

To be merged AFTER https://github.com/cultureamp/kaizen-design-system/pull/4174

### CODE REVIEW

File: packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+export const Playground: Story = {
+  render: ContextModalWithState,
+  parameters: {
+    docs: {
+      canvas: {
+        sourceState: "shown",
+      },
+    },
+  },
+  decorators: [HeightDecorator],
```
Comment (gyfchong) [inline packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx:117]: Not sure why, but adding this to the `meta` breaks the types.

File: packages/components/src/Modal/ContextModal/ContextModal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    {...props}
+  >
+    {children}
+  </ContextModal>
+)
+
+describe("<ContextModal />", () => {
+  beforeEach(() => {
+    mockMatchMedia()
+  })
```
Comment (HeartSquared) [inline packages/components/src/Modal/ContextModal/ContextModal.spec.tsx:29]: Like the other PR: Clean up after each.

File: packages/components/src/Modal/ContextModal/ContextModal.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+     * Defines the orientation layout of the image and content
+     */
+    layout?: "portrait" | "landscape"
+    title: string
+    onConfirm?: () => void
+    onDismiss: () => void
+    onAfterLeave?: () => void
+    confirmLabel?: string
+    confirmWorking?: { label: string; labelHidden?: boolean }
+    automationId?: string
```
Comment (HeartSquared) [inline packages/components/src/Modal/ContextModal/ContextModal.tsx:37]: 👋🏻

File: packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import { Decorator, Meta, StoryObj } from "@storybook/react"
+import isChromatic from "chromatic"
+import { AddImage } from "@kaizen/draft-illustration"
+import { ModalDescription } from "~components/Modal/GenericModal/subcomponents/ModalDescription"
+import { Text } from "~components/Text"
+import { ContextModal, ContextModalProps } from "../index"
+
+const IS_CHROMATIC = isChromatic()
+
+const ContextModalWithState = (args: ContextModalProps): JSX.Element => {
```
Comment (HeartSquared) [inline packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx:11]: As with the other PR (optional): Converting to a Story might be better for intellisense.

File: packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  parameters: {
+    docs: {
+      canvas: {
+        sourceState: "shown",
+      },
+    },
+  },
+}
+
+export const Potrait: Story = {
```
Comment (HeartSquared) [inline packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx:119]:
```suggestion
export const Portrait: Story = {
```

File: packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  render: ContextModalWithState,
+  args: { layout: "portrait" },
+  decorators: [HeightDecorator],
```
Comment (HeartSquared) [inline packages/components/src/Modal/ContextModal/_docs/ContextModal.stories.tsx:122]: Doesn't hurt to put the decorator into meta _unless_ only the ones with the decorator are meant to be sent to Chromatic, in which case, the chromatic parameters (most importantly `disable: false`) should only apply to these too (currently all the stories will be sent, including Playground).

File: packages/components/src/Modal/GenericModal/subcomponents/ModalContext.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { createContext } from "react"
```
Comment (HeartSquared) [inline packages/components/src/Modal/GenericModal/subcomponents/ModalContext.ts:1]: This file has been moved to `context/` right?

File: packages/components/src/utils/console.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const red = "\u001b[31m "
```
Comment (HeartSquared) [inline packages/components/src/utils/console.ts:1]: Superseded by your move :)

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**